hbase-balancer/src/main/java/org/apache/hadoop/hbase/master/balancer/PrimaryRegionCountSkewCostFunction.java

```diff
@@ -66,8 +66,12 @@ void prepare(BalancerClusterState cluster) {
   @Override
   protected void regionMoved(int region, int oldServer, int newServer) {
     cost.applyCostsChange(costs -> {
-      costs[oldServer] = computeCostForRegionServer(oldServer);
-      costs[newServer] = computeCostForRegionServer(newServer);
+      if (region == cluster.regionIndexToPrimaryIndex[region]) {
```

Comment:
Can you explain a little bit more about this change? It does not seem equal to what computeCostForRegionServer() does.

Reply:
This is to update the cost when a region is moved. The old computation recompute the cost on the src and dest server. The change just increment primary region count on the new server. If it is moved from another server, the count is decremented too for that server.

Reply:
Thanks, I got it now. Just curious, is it possible for oldServer to be < 0?

Reply:
Yes if the move is to assign a region that is currently on any server.